android/app/src/main/java/com/bitcoin/assetconverter/providers/ProviderAssetManager.kt

```diff
@@ -0,0 +1,34 @@
+package com.bitcoin.assetconverter.providers
+
+import android.content.Context
+import com.bitcoin.assetconverter.models.ProviderAssets
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+
+class ProviderAssetManager{
```

Comment:
Would it work if we just add context in the constructor 

```
class ProviderAssetManager(context: Context) {
   init {
      val json = Json { ignoreUnknownKeys = true }
        ProviderType.values().forEach {
            val assetManager = context.assets
            assetManager.open(it.resource).use { inputStream ->
                val assetMapping: ProviderAssets = json.decodeFromStream(inputStream)
                assets[it] = assetMapping.assets
            }
        }
   }
   ...
}
```

then we can just inject on a hilt module 

```
    @Singleton
    @Provides
    fun providesProviderAssetManager(@ApplicationContext appContext: Context): ProviderAssetManager =
        ProviderAssetManager(appContext)
 ```
 
 I think this `ProviderAssetManager` approach is better than the companion util I created because on the companion method I need to pass context all the time

Reply:
made the change

Reply:
Inject the context to constructor then we should be careful about memory leak, the garbage collection woudln't work. `ProviderAssetManager` may only need for Buy flow and not need to have it in singleton.

Reply:
In general all dagger injection are done by lazy. if they don't go into buy, then it won't be initialized. 

If we change the scope to viewmodel or activitiy, then we will run into the issue of constantly re-reading the json files. 
If the JSON files are small, then the re-read may not have performance impact, but if we end up having a lot of providers or a lot supported assets then re-reading is not as efficient as caching.